code/hdb/startup.q

```diff
@@ -0,0 +1,20 @@
+
+//called after new HDB cluster finishes loading all dependencies
+//triggers RDB reload instead of the HDB and removes old HDB cluster if it is a replacement
+.hdb.startup:{
+  .lg.o[`hdbstartup;"trigger rdbs to remove rows found at eod"];
+  rdbs:exec w from .servers.SERVERS where proctype = `rdb, .dotz.liveh w;
+  func:{ if[`eodtabcount in key .rdb; .rdb.reload[x]] };
+  rdbs @\: (func;.z.d);
+  .lg.o[`hdbstartup;"finished signaling rdbs to drop rows at time of eod"];
+
+  if[count deleteme:first .proc.params[`replaceProc];
+    if[count gws:exec w from .servers.SERVERS where proctype = `gateway, .dotz.liveh w;
+      .lg.o[`hdbstartup;"sending signal to delete the cluster named ",deleteme];
+      (first gws)(`.finspace.unregisterfromgw;enlist `$deleteme)
```

Comment:
why are we only sending to first gateway? what if there are multiple gateways?

is the new hdb already registered before this? it seems like this could create a period of time where there are old & new hdb(s) registered at the same time, and rdb still registered as well (I think we're clearing rdb rather than starting a fresh one, is that right?)? could open up possibilities of missing or duplicated data. we need to be careful how we do the (de)registration of all these processes - we should be simultaneously registering the new HDB *and* deregistering the old one, and need to guarantee that the rdb is cleared at the same time if we're clearing it rather than starting a new one (I'm not sure how we can make that guarantee - possibly we should also deregister the rdb while deregistering old hdb, then as part of it's "reload", have it re-register, so we will only send new queries to it after it's been cleared?)

Reply:
We need only one of the gateways to make this call to get the cluster to delete itself, but if multiple gateways, then the old hdb will remain registered to those remaining gateways.

The hdbs are (in theory) registered to the gateways on .z.po and `.hdb.startup` should get called shortly afterwards during initialization. I suppose this introduces a short window where both the new and old hdbs are registered at the same time. 

More concerning to me is the possibility of missing data if .z.po does not work as expected on finSpace and the server never gets registered (especially while the cluster is still in a "CREATING" state). If observed, that is something to probably flag to AWS, as if one node can open a handle to another node, the reverse should be able to happen as well. As a workaround, we could block any queries to the new hdb until the hdb is in a "RUNNING" state, then proceed with unblocking/registering the new hdb in the gw, then deregistering the old hdb. Still would need to address the short window of time where the new and old hdbs may be registered at the same time.